commit 91b6acfc32bee1c76727c7d3196c282cc19f3b1e
Author: Noralf Trønnes <noralf@tronnes.org>
Date:   Fri Jun 26 14:25:01 2015 +0200

    firmware: bcm2835: Support ARCH_BCM270x
    
    Support booting without Device Tree.
    Turn on USB power.
    Load driver early because of lacking support for deferred probing
    in many drivers.
    
    Signed-off-by: Noralf Trønnes <noralf@tronnes.org>
    
    firmware: bcm2835: Don't turn on USB power
    
    The raspberrypi-power driver is now used to turn on USB power.
    
    This partly reverts commit:
    firmware: bcm2835: Support ARCH_BCM270x
    
    Signed-off-by: Noralf Trønnes <noralf@tronnes.org>

diff --git a/drivers/firmware/raspberrypi.c b/drivers/firmware/raspberrypi.c
index a13558154ac3..ad915b8587fb 100644
--- a/drivers/firmware/raspberrypi.c
+++ b/drivers/firmware/raspberrypi.c
@@ -28,6 +28,8 @@ struct rpi_firmware {
 	u32 enabled;
 };
 
+static struct platform_device *g_pdev;
+
 static DEFINE_MUTEX(transaction_lock);
 
 static void response_callback(struct mbox_client *cl, void *msg)
@@ -231,6 +233,7 @@ static int rpi_firmware_probe(struct platform_device *pdev)
 	init_completion(&fw->c);
 
 	platform_set_drvdata(pdev, fw);
+	g_pdev = pdev;
 
 	rpi_firmware_print_firmware_revision(fw);
 	rpi_register_hwmon_driver(dev, fw);
@@ -245,6 +248,7 @@ static int rpi_firmware_remove(struct platform_device *pdev)
 	platform_device_unregister(rpi_hwmon);
 	rpi_hwmon = NULL;
 	mbox_free_channel(fw->chan);
+	g_pdev = NULL;
 
 	return 0;
 }
@@ -257,7 +261,7 @@ static int rpi_firmware_remove(struct platform_device *pdev)
  */
 struct rpi_firmware *rpi_firmware_get(struct device_node *firmware_node)
 {
-	struct platform_device *pdev = of_find_device_by_node(firmware_node);
+	struct platform_device *pdev = g_pdev;
 
 	if (!pdev)
 		return NULL;
@@ -280,7 +284,18 @@ static struct platform_driver rpi_firmware_driver = {
 	.probe		= rpi_firmware_probe,
 	.remove		= rpi_firmware_remove,
 };
-module_platform_driver(rpi_firmware_driver);
+
+static int __init rpi_firmware_init(void)
+{
+	return platform_driver_register(&rpi_firmware_driver);
+}
+subsys_initcall(rpi_firmware_init);
+
+static void __init rpi_firmware_exit(void)
+{
+	platform_driver_unregister(&rpi_firmware_driver);
+}
+module_exit(rpi_firmware_exit);
 
 MODULE_AUTHOR("Eric Anholt <eric@anholt.net>");
 MODULE_DESCRIPTION("Raspberry Pi firmware driver");
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1052_91b6acfc32bee1c76727c7d3196c282cc19f3b1e.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1052_91b6acfc32bee1c76727c7d3196c282cc19f3b1e.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
